lineapy/plugins/taskgen.py

```diff
@@ -0,0 +1,216 @@
+    """
+    task_definitions: Dict[str, TaskDefinition] = dict()
+
+    for session_artifacts in artifact_collection.sort_session_artifacts():
+
+        session_input_parameters_spec = (
+            BaseSessionWriter().get_session_input_parameters_spec(
+                session_artifacts
+            )
+        )
+        session_input_variables = list(session_input_parameters_spec.keys())
+        user_input_var_typing_block = [
+            f"{var} = {session_input_parameters_spec[var].value_type}({var})"
+            for var in session_input_variables
+        ]
+
+        raw_function_call_block = (
+            BaseSessionWriter().get_session_function_callblock(
+                session_artifacts, indentation=0
+            )
+        )
+        indentation_block = " " * 4
+        function_call_block = f"{indentation_block}artifacts = {pipeline_name}_module.{raw_function_call_block}"
+        return_vars = [
+            nc.safename
+            for nc in session_artifacts.usercode_nodecollections
+            if isinstance(nc, ArtifactNodeCollection)
+        ]
+
+        function_name = BaseSessionWriter().get_session_function_name(
+            session_artifacts
+        )
+
+        task_def: TaskDefinition = TaskDefinition(
+            function_name=function_name,
+            user_input_variables=session_input_variables,
+            loaded_input_variables=[],
+            typing_blocks=user_input_var_typing_block,
+            call_block=function_call_block,
+            return_vars=return_vars,
+            pipeline_name=pipeline_name,
+        )
+
+        task_definitions[function_name] = task_def
+
+    return task_definitions
+
+
+def get_allsessions_task_definition(
+    pipeline_name: str,
+) -> Dict[str, TaskDefinition]:
+    """
+    get_allsessions_task_definition returns a single task definition for the whole pipeline.
+    """
+    indentation_block = " " * 4
+    return {
+        "run_all": TaskDefinition(
+            function_name="run_all",
+            user_input_variables=[],
+            loaded_input_variables=[],
+            typing_blocks=[],
+            call_block=f"{indentation_block}artifacts = {pipeline_name}_module.run_all_sessions()"
+            "",
+            return_vars=[],
+            pipeline_name=pipeline_name,
+        )
+    }
+
+
+def get_localpickle_setup_task_definition(pipeline_name, indentation=4):
+    """
+    Returns a TaskDefinition that is used to set up pipeline that uses local pickle type
+    serialization for inter task communication.
+
+    This task should be used at the beginning of a pipeline.
+    """
+    indentation_block = " " * indentation
+    return TaskDefinition(
+        function_name="dag_setup",
+        user_input_variables=[],
+        loaded_input_variables=[],
+        typing_blocks=[],
+        call_block=f"""{indentation_block}pickle_folder = pathlib.Path('/tmp').joinpath('{pipeline_name}')
```

Comment:
If `task_local_pickle_ser/deser.jinja` are in a template file. This looks odd here.

Reply:
Yes will move and organize template file.